design-proposals/klip-43-timestamp-data-type-support.md

```diff
@@ -0,0 +1,159 @@
+# KLIP-43: TIMESTAMP Data Type Support
+
+**Author**: @jzaralim | 
+**Release Target**: 0.15 | 
+**Status**: In Discussion | 
+**Discussion**: https://github.com/confluentinc/ksql/pull/6649
+
+**tl;dr:** _KSQL currently converts time data into long values. This KLIP introduces a TIMESTAMP
+data type to handle time data instead._
+
+## Motivation and background
+
+With Connect integration, KSQL processes data from various sources and sinks. Most other databases
+have a TIMESTAMP or DATETIME type, which KSQL currently converts into long values. This sometimes
+causes problems when sinking into a database. Supporting TIMESTAMP types would make moving time data
+between KSQL and other data sources/sinks smoother and less error-prone.
+
+## What is in scope
+### Required
+
+* Add TIMESTAMP type to KSQL
+* Serialization and de-serialization of TIMESTAMPs to Avro, JSON and Delimited formats
+* Built-in functions to support TIMESTAMP usage
+
+### Future enhancements
+
+* Give window units (HOUR, DAY etc) a TIMESTAMP value
+* Add DATE and TIME types
+* UDFs such as DAY and HOUR
```

Comment:
Aren't this out of scope for this KLIP? Or are there plans to work on them? Like DATE and TIME, those seem different data types than TIMESTAMP, so they're not in this scope, right?